lisp/kbd-mode.el

```diff
@@ -150,32 +150,42 @@ If SHOW-MACROS is nil, don't highlight macros of the form
         (font-lock-add-keywords 'kbd-mode macro-regexp)
       (font-lock-remove-keywords 'kbd-mode macro-regexp))))
 
+;; How many times have it been done?
+(defun backward-symbol ()
+  "Like `forward-symbol' but backwards."
+  (forward-symbol -1))
+
 ;;; Vars
 
-(defvar kbd-mode-syntax-table nil
```

Comment:
What's the benefit here?  This seems like a syntax change for the sake of a syntax change

Reply:
Yes. I believe syntax change can have benefits, though
1. That's how it defined in the ELisp Manual and majority of built-in packages (org-mode, huge example)
=> more convenient 
2. It avoids needless repetition of code and immediately shows the content of syntax-table/map under defvar, rather than repeated variable name in docstring
2.1 The variables for which seeing docstring right below defvar do help are usually nil
2.2 C-s/r shows less results with more value
=> less clutter
3.  Everything is under one ().
=> here relying on syntax is better/easier than on semantics

Reply:
>     1. That's how it defined in the ELisp Manual and majority of built-in packages (org-mode, huge example)
>        => more convenient
>
>     2. It avoids needless repetition of code and immediately shows the content of syntax-table/map under defvar, rather than repeated variable name in docstring
>        2.1 The variables for which seeing docstring right below defvar do help are usually defined nil
>        2.2 C-s/r shows less results with more value
>        => less clutter

One may argue that, while it is slightly more clutter (and perhaps a bit more uncommon), it's also less cognitive load because one can first read the docstring and then look at the implementation (as is usually the case in LISP).

If this is the "standard" then so be it, we may have to switch to it sooner or later anyways then.  Though I'm not sure why you switched `table` to `st`; it's a short-lived local variable and both transfer roughly the same amount of information.